capirca/lib/paloaltofw.py

```diff
@@ -687,11 +624,13 @@ def _TranslatePolicy(self, pol, exp_info):
           raise PaloAltoFWUnsupportedProtocolError(
               "protocol %s is not supported" % proto_name)
 
-        # Create Term object with the term, address family, and header
-        # parameters, e.g. to/from zone, and add it to a list of
-        # terms that would form a rule.
-        new_term = Term(term, filter_type, filter_options)
-        new_terms.append(new_term)
+        if (term.icmp_type and term.protocol != ['icmp'] and
```

Comment:
nit: This will also error out for protocol ['icmp', 'icmpv6'], but the error message is not indicative of this.

From our discussion last week, we said it was okay to not have icmp and icmpv6 in the same term with icm_types. But this error message would be misleading. 

I know that NormalizeIcmpTypes also has the same behavior, but we can start with fixing this.

Reply:
Updated with additional check and error message.